src/snapred/backend/service/CalibrationService.py

```diff
@@ -307,6 +316,70 @@ def assessQuality(self, request: CalibrationAssessmentRequest):
 
         return record
 
+    @FromString
+    def normalization(self, request: NormalizationCalibrationRequest):
+        reductionIngredients = self.dataFactoryService.getReductionIngredients(request.runNumber)
+        backgroundReductionIngredients = self.dataFactoryService.getReductionIngredients(request.backgroundRunNumber)
+        runConfig = self.dataFactoryService.getRunConfig(request.runNumber)
+        calibrationRecord = self.load(runConfig)
+        calibrantSample = self.dataFactoryService.getCalibrantSample(request.calibrantPath)
+        groupingFile = request.groupingPath
+        sampleFilePath = self.dataFactoryService.getCifFilePath(request.samplePath.split("/")[-1].split(".")[0])
+        crystalInfo = CrystallographicInfoService().ingest(sampleFilePath)["crystalInfo"]
+
+        # NOTE: How do we know there aren't more than one workspace names within CalibrationRecord?
+        #       CalibrationRecord records a list of workspace names...
+        calibrationWorkspace = calibrationRecord.workspaceNames[0]
```

Comment:
I think the one you actually want is the added [here](https://github.com/neutrons/SNAPRed/blob/next/src/snapred/ui/workflow/DiffractionCalibrationCreationWorkflow.py#L124), you can identify it by comparing it to the standardized [WorkspaceName](https://github.com/neutrons/SNAPRed/blob/next/src/snapred/meta/mantid/WorkspaceNameGenerator.py#L78) for a diffcaltable.

```
difcalTableTemplate = workspaceNameGenerator.diffCalTable() \
.runNumber("")
.build()
```

and then compare each name.  But perhaps the calibRecord should be changed to have a specific member for each...?